pkg/container/container.go

```diff
@@ -298,3 +309,72 @@ func (c *containerModule) Delete(ns string, id pkg.ContainerID) error {
 
 	return container.Delete(ctx)
 }
+
+func (c *containerModule) ensureNamespace(ctx context.Context, client *containerd.Client, namespace string) error {
+	service := client.NamespaceService()
+	namespaces, err := service.List(ctx)
+	if err != nil {
+		return err
+	}
+
+	for _, ns := range namespaces {
+		if ns == namespace {
+			return nil
+		}
+	}
+
+	return service.Create(ctx, namespace, nil)
+}
+
+func setResolvConf(root string) error {
+	const tmp = "nameserver 1.1.1.1\nnameserver 1.0.0.1\n2606:4700:4700::1111\nnameserver 2606:4700:4700::1001\n"
+
+	path := filepath.Join(root, "etc/resolv.conf")
+	f, err := os.OpenFile(path, os.O_RDWR|os.O_CREATE, 644)
+	if err != nil && !os.IsNotExist(err) {
+		return err
+	}
+
+	defer f.Close()
+
+	if os.IsNotExist(err) {
+		_, err = f.WriteString(tmp)
+		return err
+	}
+
+	info, err := f.Stat()
+	if err != nil {
+		return err
+	}
+	if info.Size() == 0 {
+		_, err = f.WriteString(tmp)
+	}
+	return err
+}
+
+func applyStartup(data *pkg.Container, path string) error {
+	f, err := os.Open(path)
+	if err == nil {
```

Comment:
instead of indenting all the method implementation, it would better if u do 
```go
if err != nil {
    return err
}
defer f.Close()
// everything else goes here.
```
no the method just look weird can confusing.

Reply:
Well this is on purpose cause I really just want to do something when the file exists.
If it doesn't the this function does nothing. I though having it in the `err == nil` block would make this more obvious.

I can change it to a `os.IsNotExists` check if you prefer.

Reply:
Oh, i see. in that case i think it's fine.